### ISSUE TITLE
Uncaught TypeError: Cannot read property '1' of null

### ISSUE THREAD

**Issue by tanjapm** [James-Yu/Atom-LaTeX#129]
Decided today to give Atom a try with the intention to replace my current editor... so this error has a high potential of being my very own mistake, but I am glad for any hint or comment on where to seek and on how to potentially solve this issue...

I created a MWE:
```
\documentclass{article}
\usepackage{filecontents}

\begin{filecontents}{\jobname.bib}
  @book{Knu86,
    author = {Knuth, Donald E.},
    year = {1986},
    title = {The \TeX book},
  }
\end{filecontents}

\begin{document}

This should work. Why doesn't it?  \cite{Knu86}

\bibliographystyle{alpha}
\bibliography{\jobname}
\end{document}
```

1. I set the home directory by clicking on the home button in the status bar
2. I selected: Packages > Atom-LaTeX > Build LaTeX from active editor
3. I receive the error message and no output file

**Atom**: 1.23.0 x64
**Electron**: 1.6.15
**OS**: Microsoft Windows 10 Pro
**Thrown From**: [atom-latex](https://github.com/James-Yu/Atom-LaTeX) package 0.8.2

### Stack Trace

Uncaught TypeError: Cannot read property '1' of null

```
At %USERPROFILE%\.atom\packages\atom-latex\lib\builder.coffee:189

TypeError: Cannot read property '1' of null
    at Builder.module.exports.Builder.custom_toolchain (/packages/atom-latex/lib/builder.coffee:189:88)
    at Builder.module.exports.Builder.setCmds (/packages/atom-latex/lib/builder.coffee:155:10)
    at Builder.module.exports.Builder.build (/packages/atom-latex/lib/builder.coffee:16:6)
    at /packages/atom-latex/lib/main.coffee:34:53)
    at CommandRegistry.handleCommandEvent (~/AppData/Local/atom/app-1.23.0/resources/app/src/command-registry.js:388:36)
    at CommandRegistry.dispatch (~/AppData/Local/atom/app-1.23.0/resources/app/src/command-registry.js:273:23)
    at AtomEnvironment.module.exports.AtomEnvironment.dispatchApplicationMenuCommand (~/AppData/Local/atom/app-1.23.0/resources/app/src/atom-environment.js:1393:34)
    at EventEmitter.outerCallback (~/AppData/Local/atom/app-1.23.0/resources/app/src/application-delegate.js:335:31)
    at emitThree (events.js:116:13)
    at EventEmitter.emit (events.js:194:7)
```

### Commands

```
     -1:30.1.0 core:cancel (input.hidden-input)
     -1:29.4.0 editor:consolidate-selections (input.hidden-input)
     -1:29.4.0 core:cancel (input.hidden-input)
     -1:29.1.0 editor:consolidate-selections (input.hidden-input)
     -1:29.1.0 core:cancel (input.hidden-input)
     -1:26.1.0 command-palette:toggle (input.hidden-input)
     -1:24.6.0 editor:consolidate-selections (input.hidden-input)
     -1:24.6.0 core:cancel (input.hidden-input)
     -1:18.4.0 editor:select-line (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -1:18.1.0 fuzzy-finder:toggle-buffer-finder (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -1:14.2.0 core:move-left (input.hidden-input)
     -1:13.3.0 core:move-up (input.hidden-input)
     -1:13 core:move-down (input.hidden-input)
     -1:12.4.0 core:confirm (input.hidden-input)
     -1:05.4.0 atom-latex:build (input.hidden-input)
     -0:11.3.0 atom-latex:build-here (input.hidden-input)
```

### Non-Core Packages

```
atom-latex 0.8.2 
atom-python-run 0.9.5 
atom-runner 2.7.1 
autocomplete-python 1.10.5 
latex 0.49.0 
minimap 4.29.7 
my-word-count 0.0.0 
```

**Comment by ashthespy** [comment]
@tanjapm 
Sorry, been a bit busy the last days.
There seems to be some issue parsing the `%EXT` key in your `Custom Toolchain` - could you share what you have there?

Just another heads up - if you set the LaTeX root, then you can just use `Build` instead of `Build LaTeX from active editor`

**Comment by ashthespy** [comment]
Going to close for now, drop a line if you still are facing issues.

**Comment by kmulleners** [comment]
I'm having the same issue. If compiled in the terminal, everything is fine.
I'm using atom 1.38.2, pdf latex, custom toolchain %TEX %ARG %DOC.%EXT resulting in
pdflatex -synctex=1 -interactionmode=nonstopmode -shell-escape -file-line-error main.tex

**Comment by ashthespy** [comment]
@kpunt  Could you share your error/stack trace?

**Comment by kpunt** [comment]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.38.2 x64
**Electron**: 2.0.18
**OS**: Mac OS X 10.14.5
**Thrown From**: [atom-latex](https://github.com/James-Yu/Atom-LaTeX) package 0.8.6


### Stack Trace

Uncaught TypeError: Cannot read property '1' of null

```
At /Users/mullener/.atom/packages/atom-latex/lib/parser.coffee:95

TypeError: Cannot read property '1' of null
    at Parser.module.exports.Parser.parseLatex (/packages/atom-latex/lib/parser.coffee:95:85)
    at Parser.module.exports.Parser.parse (/packages/atom-latex/lib/parser.coffee:31:8)
    at Builder.module.exports.Builder.postBuild (/packages/atom-latex/lib/builder.coffee:112:19)
    at Builder.module.exports.Builder.buildProcess (/packages/atom-latex/lib/builder.coffee:45:8)
    at /packages/atom-latex/lib/builder.coffee:77:10)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)
```

### Commands

```
     -6:24.1.0 tree-view:toggle (input.hidden-input)
  2x -5:54.2.0 editor:toggle-line-comments (input.hidden-input)
     -5:51.3.0 atom-latex:build (input.hidden-input)
     -5:37.7.0 editor:toggle-line-comments (input.hidden-input)
  2x -5:35.1.0 core:backspace (input.hidden-input)
     -5:34.1.0 atom-latex:build (input.hidden-input)
  9x -5:14.4.0 core:backspace (input.hidden-input)
     -5:12.1.0 atom-latex:build (input.hidden-input)
     -4:36.4.0 core:move-left (input.hidden-input)
     -4:32 atom-latex:build (input.hidden-input)
  2x -4:18 atom-latex:clean (input.hidden-input)
  2x -4:11.4.0 tree-view:remove (span.name.icon.icon-file-text)
     -2:59.9.0 editor:toggle-line-comments (input.hidden-input)
     -2:59 core:save (input.hidden-input)
     -1:41.5.0 settings-view:open (input.hidden-input)
     -0:10.5.0 atom-latex:build (input.hidden-input)
```

### Non-Core Packages

```
atom-latex 0.8.6
language-latex 1.2.0
latex 0.50.2
pdf-view 0.71.0
terminal-plus 0.14.5
```


On 23 Jul 2019, at 15:06, Ash <notifications@github.com<mailto:notifications@github.com>> wrote:


@kpunt<https://github.com/kpunt> Could you share your error/stack trace?

**Comment by ashthespy** [comment]
@kpunt Could you also share what your `pdflatex` build log looks like? 
Seem to be hitting an issue with the new log parser from (https://github.com/James-Yu/Atom-LaTeX/pull/171)

**Comment by kpunt** [comment]
Please find the log-file attached.



On 23 Jul 2019, at 15:26, Ash <notifications@github.com<mailto:notifications@github.com>> wrote:


@kpunt<https://github.com/kpunt> Could you also share what your pdflatex build log looks like?
Seem to be hitting an issue with the new log parser from (#171<https://github.com/James-Yu/Atom-LaTeX/pull/171>)

**Comment by ashthespy** [comment]
@kpunt  Doesn't seem to work via email..

**Comment by kpunt** [comment]
[figurematter.log](https://github.com/James-Yu/Atom-LaTeX/files/3422370/figurematter.log)

**Comment by ashthespy** [comment]
@kpunt should be fixed with `atom-latex@v0.8.7` let me know!

**Comment by kpunt** [comment]
It is fixed indeed!
Thank you very much!

On 24 Jul 2019, at 14:58, Ash <notifications@github.com<mailto:notifications@github.com>> wrote:


@kpunt<https://github.com/kpunt> should be fixed with atom-latex@v0.8.7 let me know!
